src/app/pages/storage2/modules/devices/components/devices/devices.component.ts

```diff
@@ -53,40 +54,61 @@ export class DevicesComponent implements OnInit {
       .subscribe(() => this.loadTopologyAndDisks());
   }
 
-  private createDataSource(disks: VDev[]): void {
-    this.dataSource = new IxNestedTreeDataSource(disks);
-    this.dataSource.filterPredicate = (disks, query = '') => {
-      return disks.map((disk) => {
-        return findInTree([disk], (vdev) => {
-          switch (vdev.type) {
+  private createDataSource(dataNodes: DeviceNestedDataNode[]): void {
+    this.dataSource = new IxNestedTreeDataSource(dataNodes);
+    this.dataSource.filterPredicate = (dataNodes, query = '') => {
+      return dataNodes.map((dataNode) => {
+        return findInTree([dataNode], (dataNode) => {
+          switch (dataNode.type) {
             case VDevType.Disk:
-              return vdev.disk.toLowerCase().includes(query.toLowerCase());
+              return dataNode.disk?.toLowerCase().includes(query.toLowerCase());
             case VDevType.Mirror:
-              return vdev.name.toLowerCase().includes(query.toLowerCase());
+              return dataNode.name?.toLowerCase().includes(query.toLowerCase());
           }
         });
       }).filter(Boolean);
     };
   }
 
+  private createDataNodes(topology: PoolTopology): DeviceNestedDataNode[] {
+    const dataNodes: DeviceNestedDataNode[] = [];
+    if (topology.data.length) {
+      dataNodes.push({ children: topology.data, disk: 'Data VDEVs', guid: 'data' } as DeviceNestedDataNode);
```

Comment:
Interface fields are "the same" only because they've been artificially made to be the same. Like here, a `disk` field doesn't really make sense for an arbitrary group of vdevs.
At the very list we should use `name` here, but ideally for `DeviceNestedDataNode` is to be typed as 
```
type DeviceNestedDataNode = VdevGroup | Vdev;
```
where VdevGroup only has fields used in a group.
Or something similar to help communicate to other developers that these are really different things.

Not a problem of this PR, but `Vdev` is a misleading name as well and we should fix it in the future. What we currently call a vdev is either a vdev or a disk in a vdev with different set of attributes.

Reply:
I have addressed that in the [facbb39](https://github.com/truenas/webui/pull/6861/commits/facbb398c79b7af98e40a72e9b23df762cb54e86) commit

Reply:
@undsoft